### PR TITLE
feat: github actions details view

### LINKED ISSUE #1522: github-actions: show more details about builds
Extend the API for the github-actions-backend plugin, and extend the github-actions frontend plugin, to be able to show richer details about builds. This ticket is not specifying exactly what details to show; it has not yet been decided how much of GitHub's own GUI that we want to replicate. But individual steps of the build and their runtimes can be of interest, for example - if not the entire log output.

### PR DESCRIPTION
**Pull request by Nek**
## Hey, I just made a Pull Request!

Closes #1522 

#### :heavy_check_mark: Checklist

- [x] All tests are passing `yarn test`
- [x] Screenshots attached (for UI changes)
- [ ] Relevant documentation updated
- [x] Prettier run on changed files
- [ ] Tests added for new functionality
- [ ] Regression tests added for bug fixes
image
image
image

### CODE REVIEW

**Comment by Nek** [comment]
- [ ] error handling

File: plugins/github-actions/src/components/WorkflowRunDetailsPage/useWorkflowRunJobs.ts [hunk 1]
```diff
@@ -13,5 +13,14 @@
...
  * limitations under the License.
  */
+import { useAsync } from 'react-use';
+import { Jobs } from '../types';
 
-export { BuildInfoCard } from './BuildInfoCard';
+export const useWorkflowRunJobs = (jobsUrl?: string) => {
+  const jobs = useAsync<Jobs>(async () => {
+    if (jobsUrl === undefined) return [];
+    const data = await fetch(jobsUrl).then(d => d.json());
```
Comment (shmidt-i) [inline plugins/github-actions/src/components/WorkflowRunDetailsPage/useWorkflowRunJobs.ts:22]: Wait, shouldn't it be part of the GithubActionsApi?

File: plugins/github-actions/src/components/WorkflowRunDetailsPage/useWorkflowRunsDetails.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import { useParams } from 'react-router-dom';
+import { useAsync } from 'react-use';
+import { githubActionsApiRef } from '../../api';
+
+export const useWorkflowRunsDetails = (repo: string, owner: string) => {
+  const api = useApi(githubActionsApiRef);
+  const auth = useApi(githubAuthApiRef);
+  const { id } = useParams();
+  const details = useAsync(async () => {
+    const token = await auth.getAccessToken(['repo', 'user']);
```
Comment (shmidt-i) [inline plugins/github-actions/src/components/WorkflowRunDetailsPage/useWorkflowRunsDetails.ts:26]: is the "user" scope needed here?

File: plugins/github-actions/src/components/WorkflowRunsPage/WorkflowRunsPage.tsx [hunk 3]
```diff
@@ -22,32 +22,37 @@ import {
...
+export const WorkflowRunsPage = () => {
+  const configApi = useApi(configApiRef);
+  const repo = configApi.getString('github-actions.repo');
+  const owner = configApi.getString('github-actions.owner');
   return (
     <Page theme={pageTheme.tool}>
       <Header
         title="GitHub Actions"
-        subtitle="See recent builds and their status"
+        subtitle="See recent worflow runs and their status"
```
Comment (shmidt-i) [inline plugins/github-actions/src/components/WorkflowRunsPage/WorkflowRunsPage.tsx:41]:
```suggestion
        subtitle="See recent workflow runs and their status"
```

File: plugins/github-actions/src/components/WorkflowRunsTable/useWorkflowRuns.ts [hunk 4]
```diff
@@ -15,42 +15,58 @@
...
   const auth = useApi(githubAuthApiRef);
 
+  const errorApi = useApi(errorApiRef);
+
   const [total, setTotal] = useState(0);
   const [page, setPage] = useState(0);
   const [pageSize, setPageSize] = useState(5);
 
-  const restartBuild = async () => {};
+  const reRunWorkflow = async () => {};
```
Comment (shmidt-i) [inline plugins/github-actions/src/components/WorkflowRunsTable/useWorkflowRuns.ts:39]: Skipping this for now?

File: plugins/github-actions/src/api/GithubActionsApi.ts [hunk 5]
```diff
@@ -72,5 +72,5 @@ export type GithubActionsApi = {
     owner: string;
     repo: string;
     runId: number;
-  }) => void;
+  }) => Promise<any>;
```
[the remark this answers is not in the document]
Comment (Nek) [reply plugins/github-actions/src/api/GithubActionsApi.ts:75]: Yes

File: plugins/github-actions/package.json [hunk 6]
```diff
@@ -30,8 +30,10 @@
     "@material-ui/lab": "4.0.0-alpha.45",
     "@octokit/rest": "^18.0.0",
     "@octokit/types": "^5.0.1",
+    "moment": "^2.27.0",
     "react": "^16.13.1",
     "react-dom": "^16.13.1",
+    "react-lazylog": "^4.5.2",
```
[the remark this answers is not in the document]
Comment (Nek) [reply plugins/github-actions/package.json:37]: We don't have logs for now. When they are back it's going to be lazylog.

File: plugins/github-actions/src/components/WorkflowRunsPage/WorkflowRunsPage.tsx (hunk 3 above)
Comment (Nek) [reply plugins/github-actions/src/components/WorkflowRunsPage/WorkflowRunsPage.tsx:36]: Thanks :)

File: plugins/github-actions/src/components/WorkflowRunsTable/useWorkflowRuns.ts (hunk 4 above)
Comment (Nek) [reply plugins/github-actions/src/components/WorkflowRunsTable/useWorkflowRuns.ts:34]: I've implemented it buut there are some troubles with access rights. I'm trying to figure out how to fix it.

Comment (Nek) [reply plugins/github-actions/src/components/WorkflowRunsTable/useWorkflowRuns.ts:34]: I'm mistaken. I don't really understand where this one should be used. We already have rerun workflow button on each row of the table.

File: plugins/github-actions/src/components/WorkflowRunDetailsPage/useWorkflowRunJobs.ts (hunk 1 above)
Comment (Nek) [reply plugins/github-actions/src/components/WorkflowRunDetailsPage/useWorkflowRunJobs.ts:22]: Probably. I think I just went with the octokit flow. I'll give it another look.

Comment (Nek) [reply plugins/github-actions/src/components/WorkflowRunDetailsPage/useWorkflowRunJobs.ts:22]: I think it's fine to keep this logic outside of Api but Jobs type should be part of it.

File: plugins/scaffolder-backend/sample-templates/react-ssr-template/template.yaml [hunk 7]
```diff
@@ -22,7 +22,18 @@ spec:
         type: string
         description: Unique name of the component
       description:
-        title:  Description
+        title: Description
         type: string
         description: Description of the component
-
+---
```
Comment (shmidt-i) [inline plugins/scaffolder-backend/sample-templates/react-ssr-template/template.yaml:28]: Remove it from this file and move to a separate example in the catalog-backend? Or you can create `mock-data` script in the github-actions plugin' package.json and store this entity in the github-actions folder.

Also, this brings an interesting question to the table - if we are to allow scaffolder to create new repo with github actions out-of-the-box, how do we make sure it's going to be picked by the github-actions plugin. Like, do we hardcode `backstage.io/github-actions-id: 'spotify/backstage'` in the scaffolder sample-template. Or even import the `backstage.io/github-actions-id` as const from @backstage/plugin-github-actions? @Nek @benjdlambert  wdyt?

File: plugins/github-actions/src/components/WorkflowRunDetailsPage/useWorkflowRunsDetails.ts (hunk 2 above)
Comment (Nek) [reply plugins/github-actions/src/components/WorkflowRunDetailsPage/useWorkflowRunsDetails.ts]: No :)

File: app-config.yaml [hunk 8]
```diff
@@ -20,6 +20,10 @@ proxy:
 organization:
   name: Spotify
 
+github-actions:
```
[the remark this answers is not in the document]
Comment (Nek) [reply app-config.yaml:23]: True!

File: plugins/scaffolder-backend/sample-templates/react-ssr-template/template.yaml (hunk 7 above)
Comment (Nek) [reply plugins/scaffolder-backend/sample-templates/react-ssr-template/template.yaml:28]: On the first one, yes, definitely. I think I'll go with a separate mock-data.

File: packages/catalog-model/examples/shuffle-api-component.yaml [hunk 9]
```diff
@@ -3,6 +3,8 @@ kind: Component
 metadata:
   name: shuffle-api
   description: Shuffle API
+  annotations:
```
[the remark this answers is not in the document]
Comment (Nek) [reply packages/catalog-model/examples/shuffle-api-component.yaml]: Removing in favor of a separate mock file.

File: plugins/scaffolder-backend/sample-templates/react-ssr-template/template.yaml (hunk 7 above)
Comment (benjdlambert) [reply plugins/scaffolder-backend/sample-templates/react-ssr-template/template.yaml]: So i might have misunderstood, so bear with me...

Is it strictly necessary to store it as an annotation? if you have the source repo location already does it need to be added as an annotation?.

My thinking is that when you create a repo, and it's added to the service catalog, you should already know where the repo lives from the service definition that lives in the service catalog. Can you not query using the source URL or is the annotation there to solve a problem here?

What's the extra ID for?

Comment (Nek) [reply plugins/scaffolder-backend/sample-templates/react-ssr-template/template.yaml]: @benjdlambert but is it always the same repo? I can imagine a case when you have, let's say, a submodule which is continuously integrated only in the context of the parent project.

Comment (benjdlambert) [reply plugins/scaffolder-backend/sample-templates/react-ssr-template/template.yaml]: Yeah ok maybe that's a good enough use case

**Review by shmidt-i — APPROVED** [review]

**Pull request merged**